### DISCUSSION TITLE
Could 'brew update' auto pass through to 'brew upgrade'?

### DISCUSSION THREAD

**Discussion by tjmcewan** [Homebrew/discussions#2512] [Everyday usage]
Howdy!  Each time I go to update a package I get met with:

> Error: This command updates brew itself, and does not take formula names. Use `brew upgrade ...` instead.

Just wondering if it wouldn't be too much trouble to have it just do that?  Or perhaps update & upgrade could be synonyms - would there be conflicts?

Not exactly relevant to 90% of brew users, just a little convenience that is more human friendly.  [There's at least one other person who this affects](https://mobile.twitter.com/arnibirgisson/status/1324822370515910657). 😂 

Thanks! 🧡

**Comment by SMillerDev** [comment]
Unless you have some special settings you can just forget `brew update` exists since `brew upgrade` will already update by default.

But they're very distinct use cases so having it pass through won't work well unfortunately.

**Reply by tjmcewan** [reply]
Cool.  Thanks for the idea.  This is my brew now:
```
#!/bin/bash

if [ "$1" = "update" ]; then
  /opt/homebrew/bin/brew upgrade "${@:2}"
else
  /opt/homebrew/bin/brew "${@:1}"
fi
```

**Comment by carlocab** [comment]
See also Homebrew/brew#6125. I imagine there are other related issues.

**Reply by tjmcewan** [reply]
Ah sorry, I tried various search terms before submitting this but came up dry.
